src/lua/funcs/lua_registry.rs

```diff
@@ -25,13 +25,11 @@ pub fn index(table: AnyLuaValue, lua_key: AnyLuaValue) -> AnyLuaValue {
 // Prevent lua from changing the registry?
 pub fn new_index(table: AnyLuaValue, lua_key: AnyLuaValue, val: AnyLuaValue)
                  -> Result<(), &'static str> {
-    if let AnyLuaValue::LuaString(key) = lua_key {
-        if let Ok(json_val) = lua_to_json(val) {
-            let ref mut reg = registry::write_lock();
-            if let Some(mut reg_val) = (*reg).get_mut(&key) {
-                reg_val.set_json(json_val);
-                return Ok(());
-            }
+    if let LuaString(key) = lua_key {
+        let ref mut reg = registry::write_lock();
+        if let Some(mut reg_val) = (*reg).get_mut(&key) {
```

Comment:
unnecessary dereferencing on `reg`, and once again it might be clearer to do away with that variable altogether?